### PR TITLE
API - Unity script to support downloading call handler prompts

### PR DESCRIPTION
**Pull request by amonzerol**
See [#17159](https://github.com/Stack8/smacs/issues/17159)

### CODE REVIEW

File: ziro-export-unity.ps1 [hunk 1]
```diff
@@ -67,85 +105,89 @@ New-Item -Name "output-unity/partitions" -ItemType Directory -Force | Out-Null
...
 Write-Output "Finished getting call handlers"
 
 foreach ($CallHandler in $CallHandlers) {
     $FolderName = "callhandlers/" + $CallHandler.ObjectId
     New-Item -Name ("output-unity/" + $FolderName)  -ItemType Directory -Force | Out-Null
-    Execute-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/greetings") $Credential ($FolderName + '/greetings.json') 'Greeting' | Out-Null
-    Execute-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/transferoptions") $Credential ($FolderName + '/transferoptions.json') 'TransferOption' | Out-Null
-    Execute-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/menuentries") $Credential ($FolderName + '/menuentries.json') 'Menuentry' | Out-Null
-    Execute-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/callhandlerowners") $Credential ($FolderName + '/callhandlerowners.json') 'CallHandlerOwner' | Out-Null
+    $Greetings = Invoke-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/greetings") $Credential ($FolderName + '/greetings.json') 'Greeting'
```
Comment (justin-sadakhom) [inline ziro-export-unity.ps1:117]: I think we only call the greetings endpoint if `$CallHandler.isPrimary` is false

File: ziro-export-unity.ps1 [hunk 2]
```diff
@@ -43,10 +42,49 @@ function Execute-GetOnUnity {
...
+    ) 
+    $ProgressCount = 0
+    foreach ($Greeting in $Greetings) {
+        $ProgressCount++
+
+        $PlayWhat = [int]$Greeting.PlayWhat
+        $Enabled = [System.Convert]::ToBoolean($Greeting.Enabled)
+
+        if ($PlayWhat -eq 1 -and $Enabled -eq $true) {
+            $GreetingStreamFiles = Invoke-GetOnUnity $UnityHost ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/greetings/" + $Greeting.GreetingType + "/greetingstreamfiles") $Credential $null 'GreetingStreamFile'
```
Comment (justin-sadakhom) [inline ziro-export-unity.ps1]: Should this be using `$CallHandlerId` instead of `$CallHandler.ObjectId`?

File: ziro-export-unity.ps1 [hunk 3]
```diff
@@ -43,10 +42,49 @@ function Execute-GetOnUnity {
...
+        
+            foreach ($GreetingStreamFile in $GreetingStreamFiles) {
+                $Url = $UnityHost + ('/vmrest/handlers/callhandlers/' + $CallHandler.ObjectId + "/greetings/" + $Greeting.GreetingType + "/greetingstreamfiles/" + $GreetingStreamFile.LanguageCode + "/audio")
```
Comment (justin-sadakhom) [inline ziro-export-unity.ps1]: Same as above

File: ziro-export-unity.ps1 [hunk 4]
```diff
@@ -43,10 +42,49 @@ function Execute-GetOnUnity {
...
+
+                $Headers = @{
+                    "Accept" = "application/json"
+                }
+
+                Invoke-RestMethod -Uri $Url -Headers $Headers -SkipCertificateCheck -Credential $Credential -OutFile ("output-unity/" + $FolderName + '/gr_' + $Greeting.GreetingType + "_" + $GreetingStreamFile.LanguageCode + ".wav")
```
Comment (justin-sadakhom) [inline ziro-export-unity.ps1:81]: Question for my understanding: why is `-SkipCertificateCheck` used here?

File: ziro-export-unity.ps1 [hunk 5]
```diff
@@ -43,10 +42,49 @@ function Execute-GetOnUnity {
...
+            }
+        }
+
+        Write-Progress -activity "Getting grettings information for call handler [$CallHandlerId]..." -status "Fetched: $ProgressCount of $($Greetings.Count)" -percentComplete (($ProgressCount / $Greetings.Count) * 100)
```
Comment (justin-sadakhom) [inline ziro-export-unity.ps1]: Change `grettings` to `greetings`

**Review by justin-sadakhom — CHANGES_REQUESTED** [review]
Good work, I just had a few questions and suggestions.

File: ziro-export-unity.ps1 (hunk 4 above)
Comment (amonzerol) [reply ziro-export-unity.ps1:81]: So in the original story https://github.com/Stack8/smacs/issues/16999 it says `The script should use HTTPS and ignore certificate issues (self-signed, outdated, wrong subject name...).` and that's what it is used for

File: ziro-export-unity.ps1 (hunk 1 above)
Comment (amonzerol) [reply ziro-export-unity.ps1:117]: Right, missed that. Good catch

File: ziro-export-unity.ps1 (hunk 2 above)
Comment (amonzerol) [reply ziro-export-unity.ps1]: Indeed

**Review by justin-sadakhom — APPROVED** [review]

**Pull request merged**